File: virus.rb
class Grid
  COLUMNS = 160
  ROWS = 24

  def initialize
    @grid = Array.new(ROWS){Array.new(COLUMNS, '.')}
  end

  def display
    @grid.each { |row| puts row.join }
  end

  def infect
    @grid[rand(ROWS)][rand(COLUMNS)] = 'O'
  end

  def spread

  end
end

a = Grid.new
a.display
5.times do
  sleep(0.2)
  a.spread
  a.infect
  a.display
end

File: python_fun/virus.py
import numpy as np

class Grid():
    columns = 160
    rows = 24

    def __init__(self):
        self.grid = np.zeros([self.rows, self.columns])

    def display(self):
        for row in self.grid:
            print(''.join(row))

    def infect(self):
        np.random.choice(self.rows)
        self.grid[np.random.choice(self.rows), np.random.choice(self.columns)] = 1

g = Grid()
g.display()
g.infect()
g.display()
